src/libs/ck-libs/io/ckio.C

```diff
@@ -98,14 +136,24 @@ namespace Ck { namespace IO {
             opts.basePE = 0;
           if (-1 == opts.skipPEs)
             opts.skipPEs = CkMyNodeSize();
-
+	  if(opts.num_readers == 0){
+		// CkPrintf("DEBUG: the num readers in opts struct is 0\n");
+		opts.num_readers = std::min(32, CkNumPes());
```

Comment:
Why use 32 here? Why not 2? Why not 1 million? What does this number mean and why do you use it?

Reply:
This is more of a guess like thing, where we don't want to create too many BufferChares. It's very much a number that should be determined after tuning, so I just set 32 as a placeholder. In reality, the user probably shouldn't even let this be set.

Reply:
I would guess that a reasonable default would be something related to CmiNumNodes() or CmiNumPhysicalNodes(). Maybe different depending on whether SMP mode is being used or not.